### PR TITLE
[SeaTunnel #757][Core] Add seatunnel-core-sql module

### PR DESCRIPTION
**Pull request by leo65535**
## Purpose of this pull request

Currently, we just support basic transform framework. For more detail, please go to https://github.com/InterestingLab/seatunnel/issues/716

## Demo Job
```sql
CREATE TABLE events (
  f_type INT,
  f_uid INT,
  ts AS localtimestamp,
  WATERMARK FOR ts AS ts
) WITH (
  'connector' = 'datagen',
  'rows-per-second'='5',
  'fields.f_type.min'='1',
  'fields.f_type.max'='5',
  'fields.f_uid.min'='1',
  'fields.f_uid.max'='1000'
);

CREATE TABLE print_table (
  type INT,
  uid INT,
  lstmt TIMESTAMP
) WITH (
  'connector' = 'print',
  'sink.parallelism' = '1'
);

INSERT INTO print_table SELECT * FROM events where f_type = 1;
```

## Run Demo

1. Please download flink1.13.3 https://repo.huaweicloud.com/apache/flink/flink-1.13.3/

2. Start flink 
```
bin/start-cluster.sh
```

3. Add sql to `application.conf`
```
CREATE TABLE events (
  f_type INT,
  f_uid INT,
  ts AS localtimestamp,
  WATERMARK FOR ts AS ts
) WITH (
  'connector' = 'datagen',
  'rows-per-second'='5',
  'fields.f_type.min'='1',
  'fields.f_type.max'='5',
  'fields.f_uid.min'='1',
  'fields.f_uid.max'='1000'
);

CREATE TABLE print_table (
  type INT,
  uid INT,
  lstmt TIMESTAMP
) WITH (
  'connector' = 'print',
  'sink.parallelism' = '1'
);

INSERT INTO print_table SELECT * FROM events where f_type = 1;
```

## Demo SQL

image

image

## Check list

* [ ] Code changed are covered with tests, or it does not need tests for reason:
* [ ] Change does not need document change, or I will submit document change to https://github.com/InterestingLab/seatunnel-docs later

### CODE REVIEW

**Comment by leo65535** [comment]
cc @garyelephant @dailidong @CalvinKirs, please take a review when you are free, thanks.

File: seatunnel-core/seatunnel-core-sql/src/main/java/io/github/interestinglab/waterdrop/core/sql/splitter/SqlStatementSplitter.java [hunk 1]
```diff
@@ -0,0 +1,80 @@
...
+     * Remove comment lines.
+     */
+    private static String normalizeLine(List<String> buffer) {
+        return buffer.stream()
+            .map(statementLine -> statementLine.replaceAll(BEGINNING_COMMENT_MASK, ""))
+            .collect(Collectors.joining("\n"));
+    }
+
+    private static boolean isEndOfStatement(String line) {
+        return line.replaceAll(COMMENT_MASK, "").trim().endsWith(";");
```
Comment (kezhenxu94) [inline seatunnel-core/seatunnel-core-sql/src/main/java/io/github/interestinglab/waterdrop/core/sql/splitter/SqlStatementSplitter.java:78]: This is not safe, if the statement contains `; --` itself, the statement will be ended too early, for example, `line` is `select * from logs where log_content like "%;--count%;"`

Comment (leo65535) [reply seatunnel-core/seatunnel-core-sql/src/main/java/io/github/interestinglab/waterdrop/core/sql/splitter/SqlStatementSplitter.java:78]: In fact, will not ended too early, we will split `\n` first, give you a demo

image

**Comment by timothy1024** [comment]
+1, great job!!

**Comment by kezhenxu94** [comment]
LGTM

**Review by CalvinKirs — APPROVED** [review]
LGTM

**Pull request merged**